Services/FileLocator/FileLocatorService.cs

```diff
@@ -4,18 +4,58 @@ namespace Tanzu.Toolkit.VisualStudio.Services.FileLocator
 {
     public class FileLocatorService : IFileLocatorService
     {
-        const string _defaultCfExeName = "cf.exe";
-        const string _defaultCfCliDir = "Resources";
+        const string _cfCliV6ExeName = "cf6.exe";
+        const string _cfCliV6Dir = "Resources";
+        const string _cfCliV7ExeName = "cf7.exe";
+        const string _cfCliV7Dir = "Resources";
         const string _defaultLogsFileName = "toolkit-diagnostics.log";
         const string _defaultLogsDir = "Logs";
+        private int cliVersion = 7;
+        private readonly string pathToCf6Exe;
+        private readonly string pathToCf7Exe;
+        private readonly string _vsixBaseDirPath;
+
+        public FileLocatorService(string vsixBaseDirPath)
+        {
+            _vsixBaseDirPath = vsixBaseDirPath;
+
+            pathToCf7Exe = Path.Combine(VsixPackageBaseDir, _cfCliV7Dir, _cfCliV7ExeName);
+            pathToCf6Exe = Path.Combine(VsixPackageBaseDir, _cfCliV6Dir, _cfCliV6ExeName);
+        }
+
+        public int CliVersion
+        {
+            get => cliVersion;
+            set
+            {
+                switch (value)
+                {
+                    case 6:
+                        cliVersion = 6;
+                        break;
+                    case 7:
+                        cliVersion = 7;
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
 
         public string FullPathToCfExe
         {
             get
             {
-                var pathToCfExe = Path.Combine(VsixPackageBaseDir, _defaultCfCliDir, _defaultCfExeName);
+                switch (cliVersion)
+                {
+                    case 6:
+                        if (File.Exists(pathToCf6Exe)) return pathToCf6Exe;
+                        break;
+                    case 7:
+                        if (File.Exists(pathToCf7Exe)) return pathToCf7Exe;
+                        break;
+                }
 
-                if (File.Exists(pathToCfExe)) return pathToCfExe;
                 return null;
```

Comment:
rather than pass `null` back here and make client classes check for `null`, would it be better to throw an exception?  In other words,are there any use cases where `null` is ok?

Reply:
Yeah an exception definitely sounds like the better option here; I think `null` was a temporary solution that we never revisited